diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -0,0 +1,588 @@
+  odometry_message.child_frame_id = odom_params_.base_frame_id;
+
+  // initialize odom values zeros
+  odometry_message.twist = geometry_msgs::msg::TwistWithCovariance(
+    rosidl_runtime_cpp::MessageInitialization::ALL);
+
+  constexpr size_t NUM_DIMENSIONS = 6;
+  for (size_t index = 0; index < 6; ++index) {
+    // 0, 7, 14, 21, 28, 35
+    const size_t diagonal_index = NUM_DIMENSIONS * index + index;
+    odometry_message.pose.covariance[diagonal_index] = odom_params_.pose_covariance_diagonal[index];
+    odometry_message.twist.covariance[diagonal_index] =
+      odom_params_.twist_covariance_diagonal[index];
+  }
+
+  // initialize transform publisher and message
+  odometry_transform_publisher_ =
+    lifecycle_node_->create_publisher<tf2_msgs::msg::TFMessage>(
+    DEFAULT_TRANSFORM_TOPIC,
+    rclcpp::SystemDefaultsQoS());
+  realtime_odometry_transform_publisher_ =
+    std::make_shared<realtime_tools::RealtimePublisher<tf2_msgs::msg::TFMessage>>(
+    odometry_transform_publisher_);
+
+  // keeping track of odom and base_link transforms only
+  auto & odometry_transform_message = realtime_odometry_transform_publisher_->msg_;
+  odometry_transform_message.transforms.resize(1);
+  odometry_transform_message.transforms.front().header.frame_id = odom_params_.odom_frame_id;
+  odometry_transform_message.transforms.front().child_frame_id = odom_params_.base_frame_id;
+
+  previous_update_timestamp_ = lifecycle_node_->get_clock()->now();
+  set_op_mode(hardware_interface::OperationMode::INACTIVE);
+  return CallbackReturn::SUCCESS;
+}
+
+CallbackReturn DiffDriveController::on_activate(const rclcpp_lifecycle::State &)
+{
+  is_halted = false;
+  subscriber_is_active_ = true;
+
+  odometry_transform_publisher_->on_activate();
+  odometry_publisher_->on_activate();
+  if (publish_limited_velocity_) {
+    limited_velocity_publisher_->on_activate();
+  }
+
+  RCLCPP_INFO(
+    lifecycle_node_->get_logger(), "Lifecycle subscriber and publisher are currently active.");
+  return CallbackReturn::SUCCESS;
+}
+
+CallbackReturn DiffDriveController::on_deactivate(const rclcpp_lifecycle::State &)
+{
+  subscriber_is_active_ = false;
+  odometry_transform_publisher_->on_deactivate();
+  odometry_publisher_->on_deactivate();
+  if (publish_limited_velocity_) {
+    limited_velocity_publisher_->on_deactivate();
+  }
+  return CallbackReturn::SUCCESS;
+}
+
+CallbackReturn DiffDriveController::on_cleanup(const rclcpp_lifecycle::State &)
+{
+  if (!reset()) {
+    return CallbackReturn::ERROR;
+  }
+
+  received_velocity_msg_ptr_ = std::make_shared<Twist>();
+  return CallbackReturn::SUCCESS;
+}
+
+CallbackReturn DiffDriveController::on_error(const rclcpp_lifecycle::State &)
+{
+  if (!reset()) {
+    return CallbackReturn::ERROR;
+  }
+  return CallbackReturn::SUCCESS;
+}
+
+bool DiffDriveController::reset()
+{
+  odometry_.resetOdometry();
+
+  while (!previous_commands_.empty()) {
+    previous_commands_.pop();
+  }
```

Comment:
Alternatively, swap with an empty container to clear the queue. See [this](https://stackoverflow.com/questions/709146/how-do-i-clear-the-stdqueue-efficiently)

Reply:
yup!